junit-jupiter-engine/src/test/java/org/junit/jupiter/api/DisplayNameGenerationTests.java

```diff
@@ -0,0 +1,245 @@
+			"TEST: test()", //
+			"TEST: test(TestInfo)", //
+			"TEST: testUsingCamelCase and also UnderScores keepingParameterTypeNamesIntact(TestInfo)", //
+			"TEST: testUsingCamelCase and also UnderScores()", //
+			"TEST: testUsingCamelCaseStyle()");
+		check(UnderscoreStyleTestCase.class, expectedDisplayNames);
+		check(UnderscoreStyleInheritedFromSuperClassTestCase.class, expectedDisplayNames);
+	}
+
+	@Test
+	void noNameGenerator() {
+		check(NoNameStyleTestCase.class, List.of( //
+			"CONTAINER: nn", //
+			"TEST: @DisplayName prevails", //
+			"TEST: nn", //
+			"TEST: nn", //
+			"TEST: nn", //
+			"TEST: nn", //
+			"TEST: nn", //
+			"TEST: nn" //
+		));
+	}
+
+	@Test
+	void checkDisplayNameGeneratedForTestingAStackDemo() {
+		check(StackTestCase.class, List.of( //
+			"CONTAINER: A new stack", //
+			"CONTAINER: A stack", //
+			"CONTAINER: After pushing an element to an empty stack", //
+			"TEST: is empty()", //
+			"TEST: is instantiated using its noarg constructor()", //
+			"TEST: peek returns that element without removing it from the stack()", //
+			"TEST: pop returns that element and leaves an empty stack()", //
+			"TEST: the stack is no longer empty()", //
+			"TEST: throws an EmptyStackException when peeked()", //
+			"TEST: throws an EmptyStackException when popped()" //
+		));
+	}
+
+	private void check(Class<?> testClass, List<String> expectedDisplayNames) {
+		var request = request().selectors(selectClass(testClass)).build();
+		var descriptors = discoverTests(request).getDescendants();
+		var sortedNames = descriptors.stream().map(this::describe).sorted().collect(toList());
+		assertLinesMatch(expectedDisplayNames, sortedNames);
+	}
+
+	private String describe(TestDescriptor descriptor) {
+		return descriptor.getType() + ": " + descriptor.getDisplayName();
+	}
+
+	// -------------------------------------------------------------------
+
+	static class NoNameGenerator implements DisplayNameGenerator {
+
+		@Override
+		public String generateDisplayNameForClass(Class<?> testClass) {
+			return "nn";
+		}
+
+		@Override
+		public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+			return "nn";
+		}
+
+		@Override
+		public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+			return "nn";
+		}
+	}
+
+	@DisplayNameGeneration(NoNameGenerator.class)
+	static abstract class AbstractTestCase {
+		@Test
+		void test() {
+		}
+
+		@Test
+		void test(TestInfo testInfo) {
+			testInfo.getDisplayName();
+		}
+
+		@Test
+		void testUsingCamelCaseStyle() {
+		}
+
+		@Test
+		void testUsingCamelCase_and_also_UnderScores() {
+		}
+
+		@Test
+		void testUsingCamelCase_and_also_UnderScores_keepingParameterTypeNamesIntact(TestInfo testInfo) {
+			testInfo.getDisplayName();
+		}
+
+		@Test
+		void test_with_underscores() {
+		}
+
+		@DisplayName("@DisplayName prevails")
+		@Test
+		void testDisplayNamePrevails() {
+		}
+	}
+
+	@DisplayNameGeneration(DisplayNameGenerator.StandardGenerator.class)
```

Comment:
:o: `DisplayNameGenerator.Standard.class` class so it's more consistent with `ReplaceUnderscores`?

Reply:
✔️